src/core/api/link_api.cpp

```diff
@@ -79,6 +79,27 @@ otError otLinkSetChannel(otInstance *aInstance, uint8_t aChannel)
     return error;
 }
 
+uint8_t otLinkGetWakeupChannel(otInstance *aInstance)
+{
+    return AsCoreType(aInstance).Get<Mac::Mac>().GetWakeupChannel();
+}
+
+otError otLinkSetWakeupChannel(otInstance *aInstance, uint8_t aChannel)
+{
+    Error     error    = kErrorNone;
+    Instance &instance = AsCoreType(aInstance);
+
+    VerifyOrExit(instance.Get<Mle::MleRouter>().IsDisabled(), error = kErrorInvalidState);
+
+    instance.Get<Mac::Mac>().SetWakeupChannel(aChannel);
```

Comment:
Should we check if the `aChannel` is a valid channel?

Reply:
Agree. Done.